samples/net/sockets/echo_server/overlay-full_network_stack.conf

```diff
@@ -0,0 +1,24 @@
+# This configuration is designed to enable feature that might be disabled by default in the network stack
+CONFIG_NET_CONFIG_NEED_IPV4=y
+CONFIG_NET_CONFIG_MY_IPV4_ADDR="192.168.11.111"
+CONFIG_NET_CONFIG_PEER_IPV4_ADDR="192.168.11.100"
+CONFIG_NET_CONFIG_MY_IPV4_GW="192.168.11.1"
+
+# Optimized networking settings for performance
+CONFIG_NET_PKT_RX_COUNT=32
+CONFIG_NET_PKT_TX_COUNT=64
+CONFIG_NET_BUF_RX_COUNT=256
+CONFIG_NET_BUF_TX_COUNT=512
+
+# enable fragmenting
+CONFIG_NET_IPV4_FRAGMENT=y
+CONFIG_NET_IPV4_FRAGMENT_MAX_COUNT=16
+CONFIG_NET_IPV4_FRAGMENT_TIMEOUT=15
+CONFIG_NET_IPV4_FRAGMENT_MAX_PKT=8
+
+CONFIG_NET_IPV6_FRAGMENT=y
+CONFIG_NET_IPV6_FRAGMENT_MAX_COUNT=16
+CONFIG_NET_IPV6_FRAGMENT_TIMEOUT=15
+CONFIG_NET_IPV6_FRAGMENT_MAX_PKT=8
```

Comment:
I have no clue how this test system works, but what the configuration here is set to is to allow up to 16 packet re-assemblies at a time, for which there are a total of 8 buffers that can be used, shared for all of these (having a higher MAX_COUNT than MAX_PKT doesn't really make sense). So if you have an MTU of 1500 (i.e. ethernet) and these settings, let's assume 1 connection, you could get 8 * 1500 byte fragments to reassemble a 12,000 byte packet, or you could get 8 * 192 byte fragments to reassemble a 1,536 byte packet, but you would not be able to have 9 fragments. Therefore you need to know what the device that is the tester is using, if for example it is using 64 byte packets and trying to test the maximum size packet of 65KB then you would need to reassemble 1024 fragments - which for a small embedded device is really not going to happen. Likewise, is the tester using multiple concurrent connections or just 1?

Reply:
below is my maxwell test configuration with echo server

```
	1. Setup net-tools for bridging
git clone https://github.com/hakehuang/net-tools.git

	./net-maxwell-setup.sh

	2. Build zephyr qemu and run

           cd zephyr
           source ./zephyr-env.sh
           west update
	cd samples/net/socket/echo_server
	west build -b qemu_x86 -- -DCONF_FILE=prj.conf -DOVERLAY_CONFIG="overlay-e1000.conf;overlay-full_network_stack.conf" 
	west build -t run

```

Reply:
Looking at the test results, the MTU ranges from 8 to 512 bytes. The question here is, how many fragments can individual test send, the reports aren't clear here, for example:
> IPv6 Fragment Fragment: 8 MTU. Order: N ... 3 2 1. Overlap: None. error

@hakehuang Any chance to get information what's the value of N, i.e. how many fragments can test send? Is this value constant for all tests, or is it determined individually per test case? I think that the only way to have reliable configuration for tests is to know what is the maximum expected number of fragments.

Reply:
@rlubs, from the netcap, looks like it wraps to `z`.